src/hooks/useConnectWallet.ts

```diff
@@ -292,25 +292,35 @@ export const useConnectWallet = () => {
     modalAccountSelect.value = true;
   };
 
-  const changeEvmAccount = async (): Promise<void> => {
-    if (!currentEcdsaAccount.value.ethereum || !window.ethereum || !isH160.value) {
+  watchEffect(() => {
+    initializeWalletAccount();
+  });
+
+  const changeEvmAccount: WatchCallback<[string, any, string, any]> = (
+    [wallet, acdsaAccount, account, h160],
+    _,
+    registerCleanup
+  ) => {
+    const provider = getEvmProvider(wallet as SupportWallet);
+    if (!acdsaAccount.ethereum || !provider || !h160) {
       return;
     }
-    window.ethereum.on('accountsChanged', async (accounts: string[]) => {
-      if (accounts[0] !== currentAccount.value) {
+
+    const handleAccountsChanged = async (accounts: string[]) => {
+      if (accounts[0] !== account) {
         await disconnectAccount();
-        await setEvmWallet(selectedWallet.value as SupportWallet);
+        await setEvmWallet(wallet as SupportWallet);
       }
+    };
+
+    provider.on('accountsChanged', handleAccountsChanged);
+
+    registerCleanup(() => {
+      provider.off('accountsChanged', handleAccountsChanged);
```

Comment:
I got this error when the app was loading

<img width="659" alt="image" src="https://user-images.githubusercontent.com/92044428/174772963-cf45527d-fa18-459b-974b-f5553c456fcf.png">
<img width="1572" alt="image" src="https://user-images.githubusercontent.com/92044428/174772999-be9fd70e-bac4-48ee-9721-71c1595b99ff.png">